### ISSUE TITLE
Build pack crashes during staging

### ISSUE THREAD

**Issue by dmikusa-pivotal** [cloudfoundry/ruby-buildpack#64]
What version of Cloud Foundry are you using?

v240

What version of the buildpack you are using?

1.6.22

If you were attempting to accomplish a task, what was it you were attempting to do?

Stage an app that has a Gemfile that references a gemspec.

Ex:

```
source 'https://rubygems.org'
ruby "2.2.3"
gem "cf-app-utils"
gem "sinatra"
gemspec path: "gems/hola"
```

What did you expect to happen?

App to stage successfully.

What was the actual behavior?

Fail.

```
Staging...
-------> Buildpack version 1.6.22
       Downloaded [file:///tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/dependencies/https___buildpacks.cloudfoundry.org_concourse-binaries_bundler_bundler-1.12.5.tgz]
-----> Compiling Ruby/Rack
sh: 1: Gemfile: not found
 !
 !
 !     [!] There was an error parsing `Gemfile`: No such file or directory @ rb_sysopen - gems/hola/hola.gemspec. Bundler cannot continue.
 !
 !     #  from ./Gemfile:3
 !     #  -------------------------------------------
 !     >  gemspec path: "gems/hola"
 !     #  -------------------------------------------
 !
/tmp/d20160830-20-wxxt3t/bundler-1.12.5/gems/bundler-1.12.5/lib/bundler.rb:358:in `read':  (Bundler::Dsl::DSLError)
[!] There was an error parsing `Gemfile`: No such file or directory @ rb_sysopen - gems/hola/hola.gemspec. Bundler cannot continue.
 #  from ./Gemfile:3
 #  -------------------------------------------
 #  gem "sinatra"
 >  gemspec path: "gems/hola"
 #  -------------------------------------------
       	from /tmp/d20160830-20-wxxt3t/bundler-1.12.5/gems/bundler-1.12.5/lib/bundler.rb:358:in `block in load_gemspec_uncached'
       	from /tmp/d20160830-20-wxxt3t/bundler-1.12.5/gems/bundler-1.12.5/lib/bundler/shared_helpers.rb:54:in `chdir'
       	from /tmp/d20160830-20-wxxt3t/bundler-1.12.5/gems/bundler-1.12.5/lib/bundler/shared_helpers.rb:54:in `block in chdir'
       	from /usr/lib/ruby/2.2.0/monitor.rb:211:in `mon_synchronize'
       	from /tmp/d20160830-20-wxxt3t/bundler-1.12.5/gems/bundler-1.12.5/lib/bundler/shared_helpers.rb:53:in `chdir'
       	from /tmp/d20160830-20-wxxt3t/bundler-1.12.5/gems/bundler-1.12.5/lib/bundler.rb:347:in `load_gemspec'
       	from /tmp/d20160830-20-wxxt3t/bundler-1.12.5/gems/bundler-1.12.5/lib/bundler/dsl.rb:59:in `gemspec'
       	from ./Gemfile:3:in `eval_gemfile'
       	from /tmp/d20160830-20-wxxt3t/bundler-1.12.5/gems/bundler-1.12.5/lib/bundler/dsl.rb:39:in `instance_eval'
       	from /tmp/d20160830-20-wxxt3t/bundler-1.12.5/gems/bundler-1.12.5/lib/bundler/dsl.rb:39:in `eval_gemfile'
       	from /tmp/d20160830-20-wxxt3t/bundler-1.12.5/gems/bundler-1.12.5/lib/bundler/dsl.rb:11:in `evaluate'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/ruby_semver_version.rb:31:in `ruby_requirement'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/ruby_semver_version.rb:7:in `initialize'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/ruby_version.rb:126:in `new'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/ruby_version.rb:126:in `update_version'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/ruby_version.rb:38:in `initialize'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/ruby.rb:187:in `new'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/ruby.rb:187:in `block in ruby_version'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
       	from /usr/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/instrument.rb:16:in `instrument'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/base.rb:44:in `instrument'
 !     #  gem "sinatra"
       	from /tmp/d20160830-20-wxxt3t/bundler-1.12.5/gems/bundler-1.12.5/lib/bundler.rb:358:in `read'
       	from /tmp/d20160830-20-wxxt3t/bundler-1.12.5/gems/bundler-1.12.5/lib/bundler.rb:357:in `load_gemspec_uncached'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/instrument.rb:17:in `block in instrument'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/base.rb:48:in `instrument'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/ruby.rb:180:in `ruby_version'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/ruby.rb:331:in `block in install_ruby'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/instrument.rb:17:in `block in instrument'
       	from /usr/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/instrument.rb:16:in `instrument'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/base.rb:48:in `instrument'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/base.rb:44:in `instrument'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/ruby.rb:330:in `install_ruby'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/ruby.rb:93:in `block in compile'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/instrument.rb:17:in `block in instrument'
       	from /usr/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/instrument.rb:16:in `instrument'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/base.rb:48:in `instrument'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/base.rb:44:in `instrument'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/ruby.rb:88:in `compile'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/bin/compile:23:in `block (2 levels) in <main>'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/base.rb:129:in `log'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/bin/compile:22:in `block in <main>'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/instrument.rb:35:in `call'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/instrument.rb:35:in `block in trace'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/instrument.rb:18:in `block (2 levels) in instrument'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/instrument.rb:40:in `yield_with_block_depth'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/instrument.rb:17:in `block in instrument'
       	from /usr/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/instrument.rb:16:in `instrument'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/lib/language_pack/instrument.rb:35:in `trace'
       	from /tmp/buildpacks/70f6243610cd596d314fe836e4c3e23b/bin/compile:18:in `<main>'
Failed to compile droplet
Exit status 223
Staging failed: Exited with status 223
```

To reproduce:

1.) Download attached sample app.
2.) Run `cf push`.
3.) Staging will fail like what's listed above.

[ruby-sample.tar.gz](https://github.com/cloudfoundry/ruby-buildpack/files/445179/ruby-sample.tar.gz)

Please confirm where necessary:
* [x] I have included a log output 
* [x] My log includes an error message 
* [x] I have included steps for reproduction

**Comment by RochesterinNYC** [comment]
We've investigated this issue and determined that it's caused by Bundler using a relative path of `./Gemfile` instead of the full path to the Gemfile in its internal DSL class. Interestingly enough, we fixed this via a PR to Bundler bundler/bundler#4585 but those changes are only in bundler `1.13.+`, which does not have a stable release out yet and is not in the ruby buildpack.

We've made a change on develop (https://github.com/cloudfoundry/ruby-buildpack/commit/30e92d9497cc40d4035d454678916aec98721d28) to fix it for the time being and should release a buildpack with the fix soon.
